tuist/tuist
# Add support for defining `PRODUCT_NAME` in the manifest

### Short description 📝

Specifying the product name will allow you to have a different target and product name. This is particularly useful when you want to support different versions of the same product. E.g. different platforms.

### Solution 📦

Expose `productName` in the manifest.

### Implementation 👩‍💻👨‍💻

- [x] Add `productName` to manifest
- [x] Update `TuistGenerator` to use new variable 

### Test Plan 🛠

- Generate `fixtures/ios_app_with_frameworks` via `tuist generate`
- Run the following:
```
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme Framework2-macOS -sdk macosx
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme Framework2-iOS -sdk iphoneos
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme Framework1 -sdk iphoneos
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme Framework1 -sdk iphonesimulator
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme App -sdk iphoneos 
xcrun xcodebuild -workspace Workspace.xcworkspace -scheme App -sdk iphonesimulator
```
- Ensure all builds succeed
_optional:_
- Open the project in Xcode
- Play with it
